src/ray/gcs/store_client/redis_store_client.cc

```diff
@@ -83,55 +64,38 @@ std::vector<std::vector<std::string>> GenCommandsBatched(
   return batched_requests;
 }
 
-std::string GetKeyFromRedisKey(const std::string &external_storage_namespace,
-                               const std::string &redis_key,
-                               const std::string &table_name) {
-  auto pos = external_storage_namespace.size() + kClusterSeparator.size() +
-             table_name.size() + kTableSeparator.size();
-  return redis_key.substr(pos, redis_key.size() - pos);
-}
-
 }  // namespace
 
 void RedisStoreClient::MGetValues(const std::string &table_name,
                                   const std::vector<std::string> &keys,
                                   const MapCallback<std::string, std::string> &callback) {
-  // The `MGET` command for each shard.
-  auto batched_commands = GenCommandsBatched("HMGET", external_storage_namespace_, keys);
+  // The `HMGET` command for each shard.
+  auto batched_commands = GenCommandsBatched(
+      "HMGET", GenRedisKey(external_storage_namespace_, table_name), keys);
   auto total_count = batched_commands.size();
   auto finished_count = std::make_shared<size_t>(0);
   auto key_value_map = std::make_shared<absl::flat_hash_map<std::string, std::string>>();
 
   for (auto &command : batched_commands) {
-    auto mget_keys = std::move(command);
-    std::vector<std::string> partition_keys(mget_keys.begin() + 2, mget_keys.end());
-    auto mget_callback = [this,
-                          table_name,
-                          finished_count,
-                          total_count,
-                          mget_keys,
-                          callback,
-                          key_value_map](const std::shared_ptr<CallbackReply> &reply) {
-      if (!reply->IsNil()) {
-        auto value = reply->ReadAsStringArray();
-        // The 0 th element of mget_keys is "MGET", so we start from the 1 th
-        // element.
-        for (size_t index = 0; index < value.size(); ++index) {
-          if (value[index].has_value()) {
-            (*key_value_map)[GetKeyFromRedisKey(
-                external_storage_namespace_, mget_keys[index + 2], table_name)] =
-                *(value[index]);
+    std::vector<std::string> partition_keys(command.begin() + 2, command.end());
```

Comment:
what is + 2 for? can you comment to c ode?

Reply:
added to comment